README.md

```diff
@@ -153,36 +152,41 @@ An error response object will have the following parameters.
 }
 ```
 
 This library also exposes functions for encoding JavaScript values into Buffers which represent the bytes that a returned on-chain.
 
-- `Functions.encodeUint256(x)` takes a positive JavaScript integer number `x` and returns a 32 byte Buffer representing `x` as a `uint256` type in Solidity.
-- `Functions.encodeInt256(x)` takes a JavaScript integer number `x` and returns a 32 byte Buffer representing `x` as a `int256` type in Solidity.
-- `Functions.encodeString(x)` takes a JavaScript string `x` and returns a Buffer representing `x` as a `string` type in Solidity.
+- `Functions.encodeUint256` takes a positive JavaScript integer number and returns a Buffer of 32 bytes representing a `uint256` type in Solidity.
+- `Functions.encodeInt256` takes a JavaScript integer number and returns a Buffer of 32 bytes representing a `int256` type in Solidity.
+- `Functions.encodeString` takes a JavaScript string and returns a Buffer representing a `string` type in Solidity.
+
+Remember, it is not required to use these encoding functions.  The JavaScript code must only return a Buffer which represents the `bytes` array that is returned on-chain.
 
 ## Modifying Contracts
 
-Client contracts which initiate a request and receive a fulfillment can be modified for specific use cases. The only requirements are that the client contract extends the `FunctionsClient` contract and the `fulfillRequest` callback function never uses more than 300,000 gas.
+Client contracts which initiate a request and receive a fulfillment can be modified for specific use cases. The only requirements are that the contract sucessfully calls *sendRequest* in the *FunctionsOracle* contract and correctly implement their own *handleOracleFulfillment* function.  At this time, the maximum amount of gas that *handleOracleFulfillment* can use is 300,000. See *FunctionsClient.sol* for details.
 
 ## Simulating Requests
 
-An end-to-end request initiation and fulfillment can be simulated using the `npx hardhat functions-simulate` command. This command will report the total estimated cost of a request in LINK using the latest on-chain gas prices. Costs are based on the amount of gas used to validate the response and call the client contract's `fulfillRequest` function, plus a flat fee. Please note that actual request costs can vary based on gas prices when a request is initiated on-chain.
+An end-to-end request initiation and fulfillment can be simulated for the default *FunctionsConsumer* contract using the `functions-simulate` command. This command will report the total estimated gas use.
+If the *FunctionsConsumer* client contract is modified, this task must also be modified to accomodate the changes. See `tasks/Functions-client/simulate` for details.
+
+**Note:** The actual gas use can vary depending on the amount of data, how much data is stored on-chain and warm or cold storage slots. See the [EVM opcode gas costs](https://ethereum.org/en/developers/docs/evm/opcodes/) for details.
```

Comment:
I don't think it's necessary to explain that. Or at least just say it can vary, without mentioning details like storage slots or opcodes.

Reply:
I'll just add a note that it can vary and they should add a buffer to whatever fulfillment gas limit they set.